xapian-core/weight/tfidfweight.cc

```diff
@@ -89,12 +128,131 @@ TfIdfWeight::TfIdfWeight(const std::string &normals, double slope, double delta)
 	need_stat(DOC_LENGTH_MAX);
 	need_stat(UNIQUE_TERMS);
     }
+    switch (normalizations[0]) {
+	case 'b':
+	    wdf_norm = WDF_NORM::BOOLEAN;
+	    break;
+	case 's':
+	    wdf_norm = WDF_NORM::SQUARE;
+	    break;
+	case 'l':
+	    wdf_norm = WDF_NORM::LOG;
+	    break;
+	case 'P':
+	    wdf_norm = WDF_NORM::PIVOTED;
+	    break;
+	case 'L':
+	    wdf_norm = WDF_NORM::LOG_AVERAGE;
+	    break;
+	default:
+	    wdf_norm = WDF_NORM::NONE;
+    }
+    switch (normalizations[1]) {
+	case 'n':
+	    idf_norm = IDF_NORM::NONE;
+	    break;
+	case 's':
+	    idf_norm = IDF_NORM::SQUARE;
+	    break;
+	case 'f':
+	    idf_norm = IDF_NORM::FREQ;
+	    break;
+	case 'P':
+	    idf_norm = IDF_NORM::PIVOTED;
+	    break;
+	case 'p':
+	    idf_norm = IDF_NORM::PROB;
+	    break;
+	default:
+	    idf_norm = IDF_NORM::TFIDF;
+    }
+    wt_norm = WT_NORM::NONE;
+}
+
+TfIdfWeight::TfIdfWeight(WDF_NORM wdf_norm_,
+			 IDF_NORM idf_norm_,
+			 WT_NORM wt_norm_)
+    : wdf_norm(wdf_norm_), idf_norm(idf_norm_), wt_norm(wt_norm_),
+      param_slope(0.2), param_delta(1.0)
+{
+    if (!((wdf_norm == WDF_NORM::NONE) || (wdf_norm == WDF_NORM::BOOLEAN) ||
+	(wdf_norm == WDF_NORM::LOG_AVERAGE) || (wdf_norm == WDF_NORM::LOG) ||
+	(wdf_norm == WDF_NORM::PIVOTED) || (wdf_norm == WDF_NORM::SQUARE)))
+	throw Xapian::InvalidArgumentError("WDF_NORM is invalid");
```

Comment:
Can the compiler allow through something that would trigger this? It feels like this is just going to increase maintenance effort without adding anything useful.

Reply:
Yes , This is avoidable. Thanks for pointing it out .